### ISSUE TITLE
After ver. 4.10.0, Nextcloud URLs do not take subpath into account

### ISSUE THREAD

**Issue by EthanLatimerGB** [alainm23/planify#1398]
**Describe the bug**
When attempting to connect with a server URL of `https://domain.name.com/nextcloud`, where a reverse proxy is used when attempting to connect with the `/nextcloud` subpath, I get a 405 error returned. _(ref. Image 1)_

Previous versions of the flatpak worked correctly, in terms of the subpath being accepted _(ref. Image 2)._ But on investigation in my NGINX logs, planify does not have the `/nextcoud/`subpath included. Instead, it tried to resolve as if the URL was `https://domain.name.com/`. _(ref. Image 3)_

Was investigating and the flatpak has this bug after hash=`27495ac202bb187fb0923984bb2439f3a25211a87225b17cecfcd6f237729b3b`. Which is ver. **4.10.0**.

**To Reproduce**
1. Go to preferences
2. Select integrations
3. Select `+` to add a new account
4. Select nextcloud
5. Enter a URL for a service that has a `/nextcloud` subpage (reverse proxy)
6. Enter login details and select login.

**Expected behavior**

**Screenshots**
Image 1
![image](https://github.com/user-attachments/assets/157e4c06-18d1-4959-a22a-d393d2e48d15)
![image](https://github.com/user-attachments/assets/5a413ec8-8a67-4f78-afb3-269054b08fa1)

Image 2 - Planify before 4.10.0 would use the correct URL
![image](https://github.com/user-attachments/assets/cc6184a4-dc6a-4885-9b64-f443f4dd0d19)

Image 3 - Planify currently resolves to this URL
![image](https://github.com/user-attachments/assets/3470667d-4d41-4f1d-94b9-e73d7ea21a29)

**Desktop (please complete the following information):**
 - OS or DE: Fedora 40 Workstation (GNOME)
 - Version: 4.10.6

**Additional context**
n/a

**Comment by alainm23** [comment]
Hi and thank you for reporting this issue, I would like to ask you to tell me which url your browser redirects to when you enter this link.

`https://domain.name.com/nextcloud/.well-known/caldav`
`https://domain.name.com/.well-known/caldav`

https://github.com/user-attachments/assets/5c8f0f63-1813-4ec1-9fa6-0e4b3a88e259

**Comment by EthanLatimerGB** [comment]
Hi Alain, on entering those two links:
![image](https://github.com/user-attachments/assets/f7689460-62f2-4aef-8b54-55417bc85106)

Upon entering `/nextcloud/.well-known/caldav` I get redirected.
[Screencast from 2024-08-07 13-33-53.webm](https://github.com/user-attachments/assets/5682ab6e-a169-4de9-964e-ec696ac5a16b)

**Comment by EthanLatimerGB** [comment]
Just in case I misconfigured my reverse proxy, here is the configuration for the /nextcloud/ redirect in NGINX:
```
server {
	server_name ethanlatimer.ddns.net;
	client_max_body_size 10G;
	client_body_buffer_size 400M;

	location /nextcloud/  {
		rewrite ^/nextcloud/(.*)$ /$1 break;
		proxy_pass http://0.0.0.0:8080;
		proxy_set_header Host $host;
	    	proxy_set_header X-Real-IP $remote_addr;
    		proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    		proxy_set_header X-Forwarded-Proto $scheme;
	} 
...
```

**Comment by alainm23** [comment]
@EthanLatimerGB  I would understand that your caldav url is this right? 
`https://domain.name.com/nextcloud/remote.php/dav/`
I will make the adjustment as Planify does not take into account the path of the url.

**Comment by EthanLatimerGB** [comment]
> @EthanLatimerGB I would understand that your caldav url is this right? `https://domain.name.com/nextcloud/remote.php/dav/` I will make the adjustment as Planify does not take into account the path of the url.

Yup, thats correct and when I resolve that hostname it shows the `This is the WebDAV interface. It can only be accessed by WebDAV clients such as the Nextcloud desktop sync client.` message.
Appreciate your time mate.

**Comment by alainm23** [comment]
The fix is available in v4.10.7, please wait for the update, it should arrive in a couple of hours.

**Comment by bipox** [comment]
I was affected by this bug too and this update fixed it, thank you
